src/v/cloud_storage_clients/util.cc

```diff
@@ -67,6 +88,14 @@ error_outcome handle_client_transport_error(
     } catch (const ss::abort_requested_exception&) {
         vlog(logger.debug, "Abort requested");
         throw;
+    } catch (const ss::nested_exception& ex) {
+        if (has_abort_or_gate_close_exception(ex)) {
```

Comment:
Just curious, do you know where this was being thrown? Just want to make sure we need to worry about unwrapping the other exception types

Reply:
also curious about this. is this something we all need to be on the look out for?

Reply:
This specific instance was during a put object request handled by the s3 client:

```
WARN  2023-05-11 22:30:33,807 [shard 0] s3 - s3_client.cc:681 - S3 request failed with error: seastar::nested_exception: seastar::abort_requested_exception (abort requested) (while cleaning up after seastar::abort_requested_exception (abort requested))
ERROR 2023-05-11 22:30:33,807 [shard 0] s3 - util.cc:71 - Unexpected error seastar::nested_exception: seastar::abort_requested_exception (abort requested) (while cleaning up after seastar::abort_requested_exception (abort requested))
```

In the same time frame there were several other put requests which seem to have failed but the error logging correctly logged them as DEBUG but the nested instance got logged as error:

```
DEBUG 2023-05-11 22:30:33,796 [shard 0] s3 - util.cc:68 - Abort requested
DEBUG 2023-05-11 22:30:33,796 [shard 0] s3 - util.cc:68 - Abort requested
DEBUG 2023-05-11 22:30:33,801 [shard 0] s3 - util.cc:68 - Abort requested
ERROR 2023-05-11 22:30:33,807 [shard 0] s3 - util.cc:71 - Unexpected error seastar::nested_exception: seastar::abort_requested_exception (abort requested) (while cleaning up after seastar::abort_requested_exception (abort requested))
DEBUG 2023-05-11 22:30:33,812 [shard 0] s3 - util.cc:68 - Abort requested
DEBUG 2023-05-11 22:30:33,817 [shard 0] s3 - util.cc:68 - Abort requested
DEBUG 2023-05-11 22:30:33,825 [shard 0] s3 - util.cc:68 - Abort requested
DEBUG 2023-05-11 22:30:33,825 [shard 0] s3 - util.cc:68 - Abort requested
DEBUG 2023-05-11 22:30:33,825 [shard 0] s3 - util.cc:68 - Abort requested
DEBUG 2023-05-11 22:30:33,832 [shard 0] s3 - util.cc:68 - Abort requested
DEBUG 2023-05-11 22:30:33,838 [shard 0] s3 - util.cc:68 - Abort requested
DEBUG 2023-05-11 22:30:33,838 [shard 0] s3 - util.cc:68 - Abort requested
DEBUG 2023-05-11 22:30:33,842 [shard 0] s3 - util.cc:68 - Abort requested
```

AFAIK the issue here is how we classify and log exceptions in `handle_client_transport_error` so it should be enough to handle this specifically in this function?

> Just want to make sure we need to worry about unwrapping the other exception types

Could you expand on this? Do you mean we should be handling the nested types differently?

Reply:
>Just want to make sure we need to worry about unwrapping the other exception types
>> Could you expand on this? Do you mean we should be handling the nested types differently?

@abhijat thanks for the response. to clarify my question: might we have other places where a nested exception is being thrown but not handled correctly by unwrapping its inner/outer components.

Reply:
@dotnwat initially as mentioned in my comment [here](https://github.com/redpanda-data/redpanda/pull/10909#discussion_r1201541776) I assumed that we only need to handle this in the cloud storage client where we classify log levels, but now thinking more about it, anywhere we handle abort requested or gate closed exceptions differently from general exceptions, we may also need to examine nested exceptions to see if they should be classified as abort requested/gate closed. 

This pattern is in a few places in the tree (for example, searching for `handle_exception_type([](const ss::abort_requested_exception` yields a few results) so I suspect there are more places where we should be unwrapping nested exceptions.

Reply:
:pray: